### PR TITLE
[Tests-Only] Implement EXPECTED_FAILURES_FILE for API acceptance tests

### PR DESCRIPTION
**Pull request by phil-davis**
with similar code to https://github.com/owncloud/ocis-reva/pull/402

- [x] add `EXPECTED_FAILURES_FILE`
- [x] adjust `EXPECTED_FAILURES_FILE` for the extra failures that currently happen here in `owncloud/ocis`
- [x] add local acceptance tests
- [x] update testing documentation
- [x] run core API tests in multiple pipelines to reduce the elapsed time

### CODE REVIEW

File: tests/acceptance/bootstrap/bootstrap.php [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<?php
+$pathToCore = \getenv('PATH_TO_CORE');
```
Comment (ownclouders) [inline tests/acceptance/bootstrap/bootstrap.php:2]: Codacy found an issue: [The use of function getenv() is discouraged](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=5940619)

File: tests/acceptance/bootstrap/bootstrap.php [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+if ($pathToCore === false) {
+    $pathToCore = "../core";
+}
+
+require_once $pathToCore . '/tests/acceptance/features/bootstrap/bootstrap.php';
```
Comment (ownclouders) [inline tests/acceptance/bootstrap/bootstrap.php:7]: Codacy found an issue: ["require_once" statement detected. File manipulations are discouraged. Concatenating is forbidden. Variables inside are insecure.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=5940619)

File: tests/acceptance/bootstrap/RevaContext.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
+<?php
+
+use Behat\Behat\Context\Context;
+use Behat\Behat\Hook\Scope\BeforeScenarioScope;
+use TestHelpers\AppConfigHelper;
+use TestHelpers\SetupHelper;
+
+require_once 'bootstrap.php';
```
Comment (ownclouders) [inline tests/acceptance/bootstrap/RevaContext.php:8]: Codacy found an issue: ["require_once" statement detected. File manipulations are discouraged.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=5940619)

File: tests/acceptance/bootstrap/RevaContext.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+     *
+     * @return void
+     * @throws Exception
+     */
+    public function setUpScenario(BeforeScenarioScope $scope) {
+        // Get the environment
+        $environment = $scope->getEnvironment();
+        // Get all the contexts you need in this context
+        $this->featureContext = $environment->getContext('FeatureContext');
+        SetupHelper::init(
```
Comment (ownclouders) [inline tests/acceptance/bootstrap/RevaContext.php:33]: Codacy found an issue: [Avoid using static access to class '\TestHelpers\SetupHelper' in method 'setUpScenario'.](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=5940619)

**Comment by phil-davis** [comment]
I have something wrong in the starlark - `yaml: unmarshal errors: line 1: cannot unmarshal !!seq into yaml.RawResource`
I will look later.

**Comment by phil-davis** [comment]
I will do some more refactoring of the starlark...

**Comment by individual-it** [comment]
I suggest to merge this only after #409

**Comment by phil-davis** [comment]
> I suggest to merge this only after #409

If #409 is ready "soon" and we do not need to change test tagging in core for it, then get it merged first.

If there is complicated test tag changes needed in core for #409 then actually merging this will mean that we can manage #409 using the expected-failures file. So that might make it easier.

**Comment by individual-it** [comment]
current adjustments in core for #409: https://github.com/owncloud/core/pull/37754

**Comment by phil-davis** [comment]
> current adjustments in core for #409: [owncloud/core#37754](https://github.com/owncloud/core/pull/37754)

yes, it will be good to go ahead with that first.

**Pull request merged**